src/prism.c

```diff
@@ -9782,7 +9782,7 @@ parse_block_parameters(
     }
 
     pm_block_parameters_node_t *block_parameters = pm_block_parameters_node_create(parser, parameters, opening);
-    if (accept1(parser, PM_TOKEN_SEMICOLON)) {
+    if (accept1(parser, PM_TOKEN_SEMICOLON) && opening->type != PM_TOKEN_NOT_PROVIDED) {
```

Comment:
I think these conditions need to be reserved since accept1 will advance the lexer

Reply:
Sorry you mean "reversed"?

Reply:
Ahh sorry yes reversed